src/test_stack.py

```diff
@@ -0,0 +1,48 @@
+"""Our test file for our stack class."""
+import pytest
+
+
+
+@pytest.fixture
+def the_stack():
+    """Stack fixture."""
+    from stack import Stack
+    the_stack = Stack([1, 2, 3, 4])
+    return the_stack
+
+
+def test_stack_push(the_stack):
+    """Test for pushing to stack."""
+    the_stack.push(5)
+    assert the_stack._newLinkedList.head.data == 5
+
+
+def test_stack_pop(the_stack):
+    """Test for popping to stack."""
+    the_stack.pop()
+    assert the_stack._newLinkedList.head.data == 3
+
+
+def test_stack_pop_except_error():
+    """Test for pop exception error and should get index errror."""
+    from stack import Stack
+    empty_stack = Stack()
+    with pytest.raises(IndexError):
```

Comment:
Your `IndexError` doesn't appear to apply to the Stack, instead referring to the internal LinkedList instead. Fix that.